### DISCUSSION TITLE
关于ElasticBF热度继承复现的想法

### DISCUSSION THREAD

**Discussion by WangTingZheng** [WangTingZheng/Paperdb#13] [Ideas]
# 主要步骤
分为两步
- Step1：收集参与Compaction的SSTable的信息
  - 在Compaction类下的[inputs数组](https://github.com/WangTingZheng/Paperdb/blob/2ae2d3bb0b1b189dc96ff30b35453e6e7c749bbd/db/version_set.h#L372)
  - 收集的信息包括SSTable的Key的范围，和该SSTable的热度
- Step2：收集构建完的，范围确定的SSTable，计算它的热度
  - 遍历参与Compaction的SSTable，如果范围有重叠，就记录下热度
  - 所有记录下的热度取一个平均，计算出新的热度
 # 困难点
热度要不要持久化？不过一般Compaction出的SSTable好像是被加入到TableCache的，不会立马被从内存中剔除。

**Comment by KKould** [comment]
热度作为Meta其中一个指标进行持久化会比较好，避免重启时热度丢失而需要重新积累热度

**Reply by WangTingZheng** [reply]
可以！以后可以加一波
